cpp/examples/interop/interop.cpp

```diff
@@ -0,0 +1,159 @@
+ */
+auto make_chars_and_offsets(std::vector<std::string> strings)
+{
+  std::vector<char> chars{};
+  std::vector<cudf::size_type> offsets(1, 0);
+  for (auto& str : strings) {
+    chars.insert(chars.end(), std::cbegin(str), std::cend(str));
+    auto const last_offset = static_cast<std::size_t>(offsets.back());
+    auto const next_offset = last_offset + str.length();
+    CUDF_EXPECTS(
+      next_offset < static_cast<std::size_t>(std::numeric_limits<cudf::size_type>::max()),
+      "Cannot use strings_column_wrapper to build a large strings column");
+    offsets.push_back(static_cast<cudf::size_type>(next_offset));
+  }
+  return std::make_tuple(std::move(chars), std::move(offsets));
+};
+
+/**
+ * @brief Convert an Arrow StringViewArray to a cudf column
+ *
+ * @param array The Arrow StringViewArray
+ * @param stream The CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ */
+std::unique_ptr<cudf::column> ArrowStringViewToCudfColumn(
+  std::shared_ptr<arrow::StringViewArray> array,
+  rmm::cuda_stream_view stream      = cudf::get_default_stream(),
+  rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource())
+{
+  // Convert the string views into chars and offsets
+  std::vector<std::string> strings;
+  for (auto i = 0; i < array->length(); i++) {
+    strings.push_back(array->GetString(i));
+  }
+  auto [chars, offsets] = make_chars_and_offsets(strings);
+
+  // Copy the chars vector to the device
+  rmm::device_uvector<char> d_chars(chars.size(), stream, mr);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(
+    d_chars.data(), chars.data(), chars.size() * sizeof(char), cudaMemcpyDefault, stream.value()));
+
+  // Copy the offsets vector to the device
+  // and wrap it in a cudf::column
+  rmm::device_uvector<cudf::size_type> d_offsets(offsets.size(), stream, mr);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(d_offsets.data(),
+                                offsets.data(),
+                                offsets.size() * sizeof(cudf::size_type),
+                                cudaMemcpyDefault,
+                                stream.value()));
+  auto offsets_col = std::make_unique<cudf::column>(std::move(d_offsets), rmm::device_buffer{}, 0);
+
+  // Create a string column out of the chars and offsets
+  return cudf::make_strings_column(
+    array->length(), std::move(offsets_col), d_chars.release(), 0, {});
+}
+
+int main(int argc, char** argv)
+{
+  std::vector<std::shared_ptr<arrow::Buffer>> data_buffers;
+  std::vector<arrow::BinaryViewType::c_type> views;
+
+  // Define the data buffers
+  auto buffer_a = arrow::Buffer::FromString("helloworldapachearrowcudfnvidia");
+  data_buffers.push_back(buffer_a);
+
+  // Define the string views
+  views.push_back(ToStringView("helloworldapache", 0, 0));
+  views.push_back(ToStringView("arrowcudfnvidia", 0, 16));
+  views.push_back(ToInlineStringView("nvidia"));
```

Comment:
Could we make more strings out of these? Also is `nvidia` added twice?
```suggestion
  views.push_back(ToStringView("hello, 0, 0));
  views.push_back(ToStringView("world", 0, 5));
  views.push_back(ToStringView("apache", 0, 10));
  views.push_back(ToStringView("arrow", 0, 16));
  views.push_back(ToStringView("cudf", 0, 21));
  views.push_back(ToInlineStringView("nvidia"));
```

Reply:
I'd like the strings to be 'readable' so make them phrases or sentences (i.e. add spaces) if you want to make them longer.